adapters/xeworks/xeworks.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/httputil"
+)
+
+// { prebid: { type: 'banner' } }
+
+type bidType struct {
+	Type string `json:"type"`
+}
+
+type bidExt struct {
+	Prebid bidType `json:"prebid"`
+}
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) buildEndpointFromRequest(imp *openrtb2.Imp) (string, error) {
+	impExtRaw := imp.Ext
+	var impExt adapters.ExtImpBidder
+
+	if err := json.Unmarshal(impExtRaw, &impExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Bidder impression extension can't be deserialized",
+		}
+	}
+
+	var xeworksExt openrtb_ext.ExtXeworks
+	if err := json.Unmarshal(impExt.Bidder, &xeworksExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Xeworks extenson can't be deserialized",
+		}
+	}
+
+	endpointParams := macros.EndpointTemplateParams{
+		Host:     xeworksExt.Env,
+		SourceId: xeworksExt.Pid,
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(openRTBRequest.Imp) == 0 {
+		return nil, []error{
+			&errortypes.BadInput{
+				Message: "Imp array can't be empty",
+			},
+		}
+	}
+
+	requests := make([]*adapters.RequestData, 0, len(openRTBRequest.Imp))
+	errs := make([]error, 0)
+
+	body, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	for _, imp := range openRTBRequest.Imp {
+		endpoint, err := a.buildEndpointFromRequest(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		request := &adapters.RequestData{
+			Method:  http.MethodPost,
+			Body:    body,
+			Uri:     endpoint,
+			Headers: headers,
+		}
+
+		requests = append(requests, request)
+	}
+
+	return requests, errs
+}
+
+func (a *adapter) MakeBids(openRTBRequest *openrtb2.BidRequest, requestToBidder *adapters.RequestData, bidderRawResponse *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if httputil.IsResponseStatusCodeNoContent(bidderRawResponse) {
+		return nil, nil
+	}
+
+	if bidderRawResponse.StatusCode == http.StatusServiceUnavailable {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Bidder unavailable. Please contact the bidder support.",
```

Comment:
This is specific to your bidder so I would suggest to be specific about the bidder name here. Like - bidder XEworks not available. Or something along those lines.